pkg/adapter/info.go

```diff
@@ -23,6 +23,9 @@ import (
 type BuilderInfo struct {
 	// Name returns the official name of the adapter.
 	Name string
+	// ShortName is used to register this type with the config system.
```

Comment:
Why don't we just require the name to be in a compatible form to begin with, rather than complicate things by having N named?